src/sentry/static/sentry/app/views/usageStats/utils.tsx

```diff
@@ -0,0 +1,87 @@
+import {DataCategory} from 'app/types';
+
+export const MILLION = 10 ** 6;
+export const BILLION = 10 ** 9;
+export const UNLIMITED = '∞';
+
+export const GIGABYTE = 10 ** 9;
+
+type formatOptions = {
+  isAbbreviated?: boolean;
+  useUnitScaling?: boolean;
+};
+
+/**
+ * This expects values from CustomerUsageEndpoint, which contains usage
+ * quantities for the data categories that we sell.
+ *
+ * Note: usageQuantity for Attachments should be in BYTES
+ */
+export function formatUsageWithUnits(
+  usageQuantity: number = 0,
+  dataCategory: DataCategory,
+  options: formatOptions = {isAbbreviated: false, useUnitScaling: false}
+) {
+  if (dataCategory !== DataCategory.ATTACHMENTS) {
+    return options.isAbbreviated
+      ? _displayNumber(usageQuantity)
+      : usageQuantity.toLocaleString();
+  }
+
+  if (options.useUnitScaling) {
+    return _formatAttachmentUnits(usageQuantity);
+  }
+
+  const usageGb = usageQuantity / GIGABYTE;
+  return options.isAbbreviated
+    ? `${_displayNumber(usageGb)} GB`
+    : `${usageGb.toLocaleString(undefined, {maximumFractionDigits: 2})} GB`;
+}
+
+/**
+ * Do not use! Exporting only for re-use getsentry.
+ * Use formatReservedWithUnits or formatUsageWithUnits instead.
+ *
+ * This function is different from sentry/utils/formatBytes. Note the
+ * difference between *a-bytes (base 10) vs *i-bytes (base 2), which means that:
+ * - 1000 megabytes is equal to 1 gigabyte
+ * - 1024 mebibytes is equal to 1024 gibibytes
+ *
+ * We will use base 10 throughout billing for attachments. This function formats
+ * quota/usage values for display.
+ *
+ * For storage/memory/file sizes, please take a look at the function in
+ * sentry/utils/formatBytes.
+ */
+export function _formatAttachmentUnits(bytes: number, u: number = 0) {
+  const units = ['B', 'KB', 'MB', 'GB', 'TB', 'PB', 'EB', 'ZB', 'YB'];
+  const threshold = 1000;
+
+  while (bytes >= threshold) {
+    bytes /= threshold;
+    u += 1;
+  }
+
+  return bytes.toLocaleString(undefined, {maximumFractionDigits: 2}) + ' ' + units[u];
+}
+
+/**
+ * Do not use! Exporting only for re-use getsentry.
+ * Use formatReservedWithUnits or formatUsageWithUnits with options.isAbbreviated to true
+ */
+export function _displayNumber(n: number) {
```

Comment:
Tony pointed out that this exists here https://github.com/getsentry/sentry/blob/53fe9f7da4003f77d2e59290f39f96c107f458d4/src/sentry/static/sentry/app/utils/formatters.tsx#L201

Reply:
This behaves differently from `formatAbbreviatedNumber`. `formatAbbreviatedNumber` always round to integers while we want to have a bit of precision for usage numbers.

Reply:
Can we just add an option to `formatAbbreviatedNumber` named `precision`?

Reply:
For this, we want billion to have 2 precision point, million to have 1 precision point and thousand to integers.

If we add a `option.precision` to `formatAbbreviatedNumber`, then we would either (1) have another wrapper method to check the values then set the precision, or (2) make `formatAbbreviatedNumber` to have `option.formatForBilling` that handles that varying precision and that's just making it do two things at once. 

As it is right now, it's a simple method that's 10 lines long and specifically for formatting usage values. I rather have 2 simple methods doing almost similar things, than a hard-to-grok method trying to do two things.

Reply:
I think I would add a n extra comment on this function describing the reason this function is different from the other one, although they are similar.